### PR TITLE
Feature/iaco 1128 support external private key in origin proof

### PR DESCRIPTION
**Pull request by tomerlevi444**

### CODE REVIEW

**Comment by danevron** [comment]
I think we should add to the documentation another subsection under signing requests.

File: KlarnaOnDemandTests/KODOriginProofSpec.m [hunk 1]
```diff
@@ -8,35 +8,71 @@
...
+    });
   });
   
-  it(@"should generate a different id for each order", ^{
-    NSString *originProofA = [[[KODOriginProof alloc] initWithAmount:3600 currency:@"SEK" userToken:@"my_token"] description];
-    NSDictionary *dataDicA  = [KODSpecHelper dataDictionaryFromOriginProof:originProofA];
-    
-    NSString *originProofB = [[[KODOriginProof alloc] initWithAmount:3600 currency:@"SEK" userToken:@"my_token"] description];
-    NSDictionary *dataDicB  = [KODSpecHelper dataDictionaryFromOriginProof:originProofB];
+  describe(@"without external private key", ^{
```
Comment (danevron) [inline KlarnaOnDemandTests/KODOriginProofSpec.m]: Did you mean with external private key?

**Comment by danevron** [comment]
I tried my best but didn't understand most of the PR. It's understandable given the different language and tools.

**Comment by tomerlevi444** [comment]
@danevron I don't think that documentation is required for this task, since it is meant only for backwards compatibility for KnowIT's use-case

**Pull request merged**